tests/Unit/Auth0UserRepositoryTest.php

```diff
@@ -0,0 +1,145 @@
+     */
+    public function testUserModelFromConfig()
+    {
+        foreach (['TestJwt', '\TestJwt', 'App\JwtFoo', '\App\JwtBar'] as $fakeClass) {
+            // Create new class, extending Auth0User.
+            $mock = Mockery::namedMock($fakeClass, Auth0User::class);
+
+            // Try both the class name itself as the class definition.
+            foreach ([$fakeClass, get_class($mock)] as $value) {
+                // Set config value to fake class.
+                Config::set('mrd-auth0.model', $value);
+
+                // Get repository from service container.
+                $repository = App::make(\Auth0\Login\Contract\Auth0UserRepository::class);
+
+                // Verify that the repository indeed has the correct model.
+                self::assertStringEndsWith($fakeClass, $repository->getModel());
+                self::assertStringStartsWith('\\', $repository->getModel());
+            }
+
+            Mockery::close();
+        }
+    }
+
+    /**
+     * Verifies that the JWT model config value is used when creating a new instance.
+     */
+    public function testJwtModelFromConfig()
+    {
+        foreach (['Test', '\Test', 'App\Foo', '\App\Bar'] as $fakeClass) {
+            // Create new class, extending Auth0User.
+            $mock = Mockery::namedMock($fakeClass, Auth0JWTUser::class);
+
+            // Try both the class name itself as the class definition.
+            foreach ([$fakeClass, get_class($mock)] as $value) {
+                // Set config value to fake class.
+                Config::set('mrd-auth0.jwt-model', $value);
+
+                // Get repository from service container.
+                $repository = App::make(\Auth0\Login\Contract\Auth0UserRepository::class);
+
+                // Verify that the repository indeed has the correct model.
+                self::assertStringEndsWith($fakeClass, $repository->getJwtModel());
+                self::assertStringStartsWith('\\', $repository->getJwtModel());
+            }
+
+            Mockery::close();
+        }
+    }
+
+    /**
+     * Verify that the getUserByUserInfo returns the correct instance.
+     */
+    public function testGetUserByUserInfo()
+    {
+        $repository = new Auth0UserRepository(Auth0User::class, Auth0JWTUser::class);
+
+        $user = $repository->getUserByUserInfo(['profile' => [], 'accessToken' => []]);
+
+        self::assertInstanceOf(Auth0User::class, $user);
+    }
+
+    /**
+     * Verify that the getUserByDecodedJWT returns the correct instance.
+     */
+    public function testGetUserByDecodedJWT()
+    {
+        $repository = new Auth0UserRepository(Auth0User::class, Auth0JWTUser::class);
+
+        $user = $repository->getUserByDecodedJWT([]);
+
+        self::assertInstanceOf(Auth0JWTUser::class, $user);
+    }
+
+    /**
+     * Verify an InvalidArgumentException is thrown when the given user class does not extend Auth0User.
```

Comment:
Why would a given user class be something else than an extension of Auth0User? Same for Auth0JWTUser.

Reply:
So when using this extension you have to specify in the config a PHP class which is going to be used by the Auth0 package. The user should always specify a custom class which extends Auth0User since otherwise the underlying Auth0 package does not work. We are just checking for robustness whether this is indeed the case, to give an error early on instead of a weird error by Auth0 which you then have to debug.